command/jsonplan/plan.go

```diff
@@ -212,10 +212,15 @@ func (p *plan) marshalResourceChanges(changes *plans.Changes, schemas *terraform
 		if err != nil {
 			return err
 		}
+		// We drop the marks from the change, as decoding is only an
+		// intermediate step to re-encode the values as json
+		changeV.Before, _ = changeV.Before.UnmarkDeep()
+		changeV.After, _ = changeV.After.UnmarkDeep()
```

Comment:
Rather than mutating the struct, it might be clearer to bind the results of the `UnmarkDeep` calls to locals inside the `if changeV.[Before|After]` != cty.NilVal` blocks below. Those are already messily referring back to the change source to retrieve the marks.

One other worry here is that I think calling `UnmarkDeep` on `cty.NilVal` might crash. I'm not completely sure that this is the case, but moving the unmarking inside the if blocks feels safer.

(Same thought for the other changes in this package.)

Reply:
I verified that `UnmarkDeep` in a noop with a nil val, and opted to overwrite the fields here to leave the rest of the code as unchanged as possible. 

I also don't particularly like mutating the value, but since the `changeV` values have no way of leaking out from this package, I went with the smaller change.